cairo/src/image_surface.rs

```diff
@@ -107,6 +107,26 @@ impl ImageSurface {
         }
     }
 
+    pub fn consume_data<'a>(self) -> Result<&'a mut [u8], BorrowError> {
```

Comment:
This is unsound. The data is owned by the surface, but the surface is destroyed after this function returned so you'll end up with a reference to already freed data.

I'm not sure how this can be implemented as the data of the surface could be of any type and we don't even know how to free it.

You could maybe create a new type that `Deref`s / `DerefMut`s into `[u8]`, keeps the reference to the image surface alive and then on `Drop` finally gets rid of the surface?

Reply:
I guess the test case works because the freed memory is not overwritten. I'll redo this, thanks for the comment.